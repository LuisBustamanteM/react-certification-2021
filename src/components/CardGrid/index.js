export { default } from './CardGrid.component';