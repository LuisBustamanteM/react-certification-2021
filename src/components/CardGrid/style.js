import styled from 'styled-components';

export const CardGrid = styled.div`
    padding: 20px;
    margin: 0px auto;
    justify-content: flex-start;
    display: flex;
    flex-flow: row wrap;
    box-sizing: border-box;
    width: 1500px;
    @media screen and (max-width: 1500px) {
        width:1135px;
    }
    @media screen and (max-width: 1200px) {
        width:770px;
    }
    @media screen and (max-width: 800px) {
        width:100%;
        -moz-box-pack: center;
        justify-content: center;
    }
`;

export const ErrorMessage = styled.div`
  text-align: center;
  font-size: 24px;
  color: #965108;
  width: 100%;
`;