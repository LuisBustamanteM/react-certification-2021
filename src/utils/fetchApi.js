async function fetchApi(url){
    let data = [];
    try{
        data = await fetch(url)
            .then(res => res.json());

        return data;

    } catch(e){
        console.log(e);
        return data;
    }

}

export default fetchApi;