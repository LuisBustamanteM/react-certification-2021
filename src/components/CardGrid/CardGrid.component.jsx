import React, {useState, useEffect} from 'react';
import VideoCard from '../VideoCard';
import {CardGrid, ErrorMessage } from './style';

const CardGridComponent = ({items = []}) => {

    return(
        <CardGrid>
            { items.length > 0 ? items.map( (item, idx) => (
                    <VideoCard key={idx}
                               id={item.id.channelId}
                               defaultImage={item.snippet.thumbnails.default.url}
                               mediumImage={item.snippet.thumbnails.medium.url}
                               highImage={item.snippet.thumbnails.high.url}
                               title={item.snippet.title}
                               description={item.snippet.description}/>
                ))
                : <ErrorMessage>No videos to display, Try Reloading.</ErrorMessage>
            }
        </CardGrid>
    )
}

export default CardGridComponent;