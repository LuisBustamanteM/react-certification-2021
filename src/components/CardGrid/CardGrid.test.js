import React, {useEffect, useState} from 'react';
import ReactDOM from  'react-dom';

import CardGrid from './index';
import {render} from "@testing-library/react";
import fetchApi from "../../utils/fetchApi";

describe("Testing <CardGrid/>", () => {

    test("<CardGrid/> matches snapshot with items",  async () => {
        let items = await fetchApi("http://localhost:8080/videos")
                            .then(data => data.items);

        const component = render (<CardGrid items={items}/>);
        expect (component.container).toMatchSnapshot();
    });

    test("<CardGrid/> matches snapshot without items",  () => {
        const component = render (<CardGrid />);
        expect (component.container).toMatchSnapshot();
    });

});