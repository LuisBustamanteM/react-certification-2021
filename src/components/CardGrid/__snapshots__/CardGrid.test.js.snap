// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CardGrid/> <CardGrid/> matches snapshot with items 1`] = `
<div>
  <div
    class="sc-jSgvzq cErgQG"
  >
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea dYaEOh"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea cOyLmM"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Video Tour | Welcome to Wizeline Guadalajara
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea biegUW"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline Guadalajara | Bringing Silicon Valley to Mexico
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Wizeline continues to offer a Silicon Valley culture in burgeoning innovation hubs like Mexico and Vietnam. In 2018, our Guadalajara team moved into a ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea llcUvq"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline hace sentir a empleados como en casa
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          En el 2014, Bismarck fundó Wizeline, compañía tecnológica que trabaja con los corporativos ofreciendo una plataforma para que desarrollen software de forma ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea jIBVJn"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          We Are Wizeline
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Engineering a better tomorrow. Wizeline is a global software development company that helps its clients solve their biggest challenges with design and ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea jVxTYl"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline Thrives in Mexico City
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Our vibrant Mexico City office is home to agile software engineers, talented UX designers, and brilliant data scientists. Learn about the rich history of Mexico City.
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea jgDoJL"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          El plan de Wizeline, una empresa de inteligencia artificial, para ayudar a crecer la comunidad de ciencia de datos en CDMX y todo el país, a través de cursos ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea dmDwxs"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          A Day in the Life of an Engineering Manager at Wizeline
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Fernando Espinoza shares his experience working as an engineering manager at Wizeline and mentoring other engineers. Learn about Fernando's passions ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea kHJWKr"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        />
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea juJhdS"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Silicon Valley en México
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Empresas de Silicon Valley buscan establecerse en México por el gran talento que hay en nuestro país. Es una investigación de Roberto Domínguez.
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea lnoXYF"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Why Wizeline? featuring Juan Pablo Villa in Mexico City
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Juan Pablo, known as Gianpa at Wizeline, is a software engineer in our Mexico City office. Gianpa focuses on Android apps, is an integral part of our culture, ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea jIDFbi"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline&#39;s 2020 Year in Review
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          There's no doubt that 2020 has been an unprecedented year. However, amidst all the chaos, we achieved remarkable growth in various areas of our business.
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea mTNvH"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Why Wizeline? featuring Hugo Lopez in Mexico City
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        />
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea cSuAnm"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Why Wizeline? (We&#39;re Hiring in Mexico!)
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          A quick look at why people join Wizeline, what motivates us as a team and what it's like to work in our Guadalajara office. Learn more and apply here: ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea NwfiX"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Why Wizeline? featuring Oswaldo Herrera in Mexico City
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Oswaldo is a software engineering in Wizeline's Mexico City office. He joined Wizeline because of the camaraderie and deep sense of commitment of our teams.
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea gRaIcD"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Caso de Éxito AWS: Wizeline [Spanish]
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Central de socios de APN - https://amzn.to/2S7tIXM Fundada en 2014, Wizeline es una compañía joven e innovadora que nació en la nube para ofrecer soporte ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea dUdCCa"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline&#39;s 2019 Year in Review
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        />
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea inRusv"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          [1 of 2] Wizeline CEO shares career lessons from Google
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Founder & CEO Bismarck Lepe on growth opportunities at Wizeline and his career-path experience as an early Google employee. Join our team!
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea iiYXKP"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline Data Engineering featuring Tania Reyes
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Tania discovered her interest in Big Data while working at Wizeline and took Wizeline Academy courses to skill up and join the data team. Now, she works on ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea cNJKkV"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          A small but mighty (and growing) team, the Queretaro crew has taken ownership of growing the office and brand, speaking at university events, hosting tech ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea bPQzjJ"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Miriam Godinez | Women in Leadership at Wizeline
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Science and technology always caught Miriam's attention. One of her ultimate goals as a Senior Engineer Manager and Lead from the Mobile Team at Wizeline ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea llMlrL"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Welcome Back to Wizeline Vietnam | Extended Version
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Thanks to swift government action, the COVID-19 situation in Vietnam has reached a point where businesses are able to return to work and reopen offices.
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea fWhIND"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          How does Wizeline work?
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Wizeline builds teams with a mix of technical and non-technical talent to deliver better products, faster. Learn more about our consulting services: ...
        </p>
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea fmLsql"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          12 Wishes from Wizeline | Happy Holidays 2019
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        />
      </div>
    </div>
    <div
      class="sc-bdfBQB hsGamh"
    >
      <div
        class="sc-gsTEea eshXBn"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          Why Wizeline?
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          Hear from our employees directly about what excites them about their roles here at Wizeline. Wizeline wants to hire the best and the brightest to accelerate their ...
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing <CardGrid/> <CardGrid/> matches snapshot without items 1`] = `
<div>
  <div
    class="sc-jSgvzq cErgQG"
  >
    <div
      class="sc-gKseQn flKMzi"
    >
      No videos to display, Try Reloading.
    </div>
  </div>
</div>
`;
